db/db.go

```diff
@@ -265,7 +265,14 @@ func IsConnectionError(err error) bool {
 	if err == nil {
 		return false
 	}
-	lowerCaseError := strings.ToLower(err.Error())
+
+	var cmdErr mongo.CommandError
+	var ok bool
+	if cmdErr, ok = err.(mongo.CommandError); !ok {
+		return false
```

Comment:
don't we want to revert to the original behavior (`strings.ToLower(err.Error())`) in the event that we can't cast?

Reply:
If these aren't all command errors, then we probably want to fallback to err.Error(). I'm not sure what kinds of errors these are.

Reply:
Changed to fallback to err.Error() in the next commit.